### ISSUE TITLE
No login with admin on HTTP

### ISSUE THREAD

**Issue by WGrobler** [boringproxy/boringproxy#158]
When trying to log in to the **admin-domain** on a server with **allow-http**, the login page is displayed correctly, but whenever you enter the correct token, the page redirects to back the login page without any error on the web UI or console.

This happens because the Cookie is set with Secure Flag upon successful authentication ([ui_handler.go:494](https://github.com/boringproxy/boringproxy/blob/b1eb2cec438b260d0ec804c6d8a53b8815692b9f/ui_handler.go#L494)). Since the request was done over HTTP, the cookie is not set and no error is displayed. Normally after the cookie is set, the page redirects to /tunnels, but since there is no access token cookie, the web UI redirects again to the login page. 

I agree with the reasoning behind setting the cookie to secure and accessing the admin panel over HTTP is not a good idea (in production environments). However, because the admin domain can be accessed over HTTP and there is no error message, it causes some confusion.

Either accessing the **admin-domain** should be blocked when using HTTP, even if the **allow-http** flag is set or the Cookie must be set to Secure:False when accessing via HTTP. 

I don't know if the  Secure:False cookie will be overwritten when accessing the **admin-domain** again using HTTPS. If you are working with standard ports (80 & 443) the insecure cookie set using HTTP should be accessible from HTTPS, but it will send the cookie information in plan text and this creates a new security issue. In this instance the set cookie needs to be investigated, if determined that it's unsecure, it should be removed and updated with a secure cookie. This introduces a lot of complexity and the easiest option would probably be to disable **admin-domain** access over HTTP altogether.

**Comment by anderspitman** [comment]
I agree, HTTP access to the admin domain should be blocked.

**Comment by retiolus** [comment]
So, how can we connect to an admin-domain that is not using ports 80/443?

**Comment by anderspitman** [comment]
@retiolus I don't think this is currently possible. That would send a security cookie unencrypted over the internet, which is not a good idea. You need to either use the built-in TLS cert capabilities of boringproxy, or put it behind another reverse proxy which handles TLS termination.
